src/webviews/extension-side/plotView/rendererCommunication.node.ts

```diff
@@ -2,61 +2,16 @@
 // Licensed under the MIT License.
 
 import { inject, injectable } from 'inversify';
-import { Event, extensions, NotebookEditor, window } from 'vscode';
-import { IExtensionSingleActivationService } from '../../../platform/activation/types';
-import { disposeAllDisposables } from '../../../platform/common/helpers';
-import { IDisposable } from '../../../platform/common/types';
-import { noop } from '../../../platform/common/utils/misc';
 import { PlotSaveHandler } from './plotSaveHandler.node';
-import { PlotViewHandler } from './plotViewHandler.node';
-
-export type OpenImageInPlotViewer = {
-    type: 'openImageInPlotViewer';
-    outputId: string;
-    mimeType: string;
-};
-export type SaveImageAs = {
-    type: 'saveImageAs';
-    outputId: string;
-    mimeType: string;
-};
+import { PlotViewHandler } from './plotViewHandler';
+import { RendererCommunication as RendererCommunicationBase } from './rendererCommunication';
 
 @injectable()
-export class RendererCommunication implements IExtensionSingleActivationService, IDisposable {
-    private readonly disposables: IDisposable[] = [];
+export class RendererCommunication extends RendererCommunicationBase {
     constructor(
-        @inject(PlotSaveHandler) private readonly plotSaveHandler: PlotSaveHandler,
-        @inject(PlotViewHandler) private readonly plotViewHandler: PlotViewHandler
-    ) {}
-
-    public dispose() {
-        disposeAllDisposables(this.disposables);
-    }
-    public async activate() {
-        const ext = extensions.getExtension('ms-toolsai.jupyter-renderers');
-        if (!ext) {
-            return;
-        }
-        if (!ext.isActive) {
-            await ext.activate();
-        }
-        const api = ext.exports as {
-            onDidReceiveMessage: Event<{ editor: NotebookEditor; message: OpenImageInPlotViewer | SaveImageAs }>;
-        };
-        api.onDidReceiveMessage(
-            ({ editor, message }) => {
-                const document = editor.notebook || window.activeNotebookEditor?.notebook;
-                if (!document) {
-                    return;
-                }
-                if (message.type === 'saveImageAs') {
-                    this.plotSaveHandler.savePlot(document, message.outputId, message.mimeType).catch(noop);
-                } else if (message.type === 'openImageInPlotViewer') {
-                    this.plotViewHandler.openPlot(document, message.outputId).catch(noop);
-                }
-            },
-            this,
-            this.disposables
-        );
+        @inject(PlotSaveHandler) plotSaveHandler: PlotSaveHandler,
+        @inject(PlotViewHandler) plotViewHandler: PlotViewHandler
+    ) {
+        super(plotSaveHandler, plotViewHandler);
```

Comment:
Curious why do we need this class? Can't we just use the same class from `rendererCommunication.ts` file?

Reply:
I think I can get it to work as you say. I'll take a look! Thank you!

Reply:
How does this look? https://github.com/microsoft/vscode-jupyter/pull/10881/commits/c612ebf36eb62f5656abfdb7626ef056cf47fda8